README.md

```diff
@@ -13,43 +16,196 @@ This project is still heavily under development and is not ready for use
+
+This guide will configure cert-manager to attempt to fulfill Certificate
+resources in all namespace in your cluster.
+
+### 0. Pre-requisites
 
 * Kubernetes cluster with CustomResourceDefinitions enabled (1.7+) (see [#49](https://github.com/jetstack-experimental/cert-manager/issues/49))
 
-## Deploying
+### 1. Deploy cert-manager
 
-To deploy the latest development version, run:
+To deploy the latest version of cert-manager, run:
 
 ```
-$ kubectl create -f docs/cert-manager.yaml
+$ kubectl create -f https://github.com/jetstack-experimental/cert-manager/blob/master/docs/cert-manager.yaml
 ```
 
-## Getting started
+In future this may be replaced with a Helm chart. There are currently no
+official RBAC roles defined for cert-manager (see [#34](https://github.com/jetstack-experimental/cert-manager/issues/34))
+
+### 2. Set up letsencrypt staging issuer
+
+An Issuer in cert-manager describes a source for signed TLS certificates that
+cert-manager can use to fulfil Certificate resources in a Kubernetes cluster.
+
+Within the Issuers spec, we can define any configuration that may be required
+(eg. credentials for updating a DNS server) on a per-issuer basis.
+
+In the below example, you **must** remember to fill in the `spec.acme.email`
+field.
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: ""
+    # Name of a secret used to store the ACME account private key
+    privateKey: letsncrypt-staging
+    # ACME dns-01 provider configurations
+    dns-01:
+      # Here we define a list of DNS-01 providers that can solve DNS challenges
+      providers:
+      # We define a provider named 'prod-dns', with configuration for the
+      # clouddns challenge provider.
+      - name: prod-dns
+        clouddns:
+          # A secretKeyRef to a the google cloud json service account
+          serviceAccount:
+            name: clouddns-service-account
+            key: service-account.json
+          # The project in which to update the DNS zone
+          project: gcloud-prod-project
+```
 
-To get started, I've created two example issuers in `docs/acme-issuer.yaml`.
-These are configured to support the clouddns challenge provider for ACME, but
-if you do not intend to test this functionality then feel free to remove the
-configuration for it.
+This is an example Issuer for the letsencrypt staging server. Here, we define
+one DNS provider, named clouddns, that can be used to solve ACME challenges.
+
+HTTP-01 is also supported without additional configuration when using the ACME
+issuer.
+
+Upon creation of the Issuer, any initial preparation for that Issuer will be
+performed, e.g. for the ACME issuer, an account is registered with the ACME
+server specified and a corresponding private key generated too if required.
```

Comment:
❓ Maybe this could be clearer....perhaps: "...registered on the *chosen* ACME server and a private key will be generated too, if required."